### PR TITLE
feat: added ToDo list

### PR DESCRIPTION
**Pull request by TejasShekar**
- Added todo list 
- todo data will stored in local storage for persist
- delete todo after completed.

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -111,7 +125,63 @@ p {
   bottom: 1rem;
 }
 
-@media only screen and (min-device-width: 360px) and (max-device-width: 812px) {
+.toggle-todo {
+  background-color: transparent;
+  color: rgb(201, 201, 201);
```
Comment (krgarima) [inline src/App.css:130]: use css var names

**Review by krgarima — APPROVED** [review]
Good to merge

**Review by arpit01923 — APPROVED** [review]
- clean code
- good folder strcture and naming convention
- one suggestion, try barrel export for context and reducer

File: src/App.css [hunk 2]
```diff
@@ -111,7 +125,63 @@ p {
...
+  font-size: 1rem;
+}
+
+.toggle-todo:hover {
+  color: rgb(255, 255, 255);
```
Comment (pariyar07) [inline src/App.css:135]: If you have a color variable use that, makes it consistent

File: src/components/HomePage.jsx [hunk 3]
```diff
@@ -1,15 +1,20 @@
+import { TodoProvider } from "../context/todo-context";
```
Comment (pariyar07) [inline src/components/HomePage.jsx:1]: try absolute imports

File: src/components/HomePage.jsx [hunk 4]
```diff
@@ -1,15 +1,20 @@
...
 import { GreetUserWithTime } from "./GreetUserWithTime";
 import { MainFocus } from "./MainFocus";
 import { Quotes } from "./Quotes";
+import { ToDo } from "./ToDo";
 import { Weather } from "./Weather";
```
Comment (pariyar07) [inline src/components/HomePage.jsx:6]: use barrel exports

File: src/components/ToDo.jsx [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    <div className="flex flex-col todo-container">
+      <div
+        className="todo-list flex-col px-1 py-1"
+        style={{
+          display: `${isTodoOpen ? "flex" : "none"}`,
+        }}
+      >
+        <ol className="mb-sm" style={{ overflow: "auto" }}>
+          {todoList.length !== 0 ? (
+            todoList.map(({ id, todo, checked }, index) => (
```
Comment (pariyar07) [inline src/components/ToDo.jsx:43]: great that you have destructured here

**Review by pariyar07 — COMMENTED** [review]
Didn't see any important changes to make but look at the suggestion given. The code looks readable and clean.

File: src/App.css (hunk 1 above)
Comment (TejasShekar) [reply src/App.css:130]: I'm not repeating it anywhere. So, I did not use variables for the colors. But I have used color variables for repeating colors.

**Pull request merged**